docs/en/sql-reference/functions/other-functions.md

```diff
@@ -660,20 +660,45 @@ SELECT
 
 ## formatReadableTimeDelta
 
-Given a time interval (delta) in seconds, this function returns a time delta with year/month/day/hour/minute/second as string.
+Given a time interval (delta) in seconds, this function returns a time delta with year/month/day/hour/minute/second/millisecond/microsecond/nanosecond as string.
 
 **Syntax**
 
 ``` sql
-formatReadableTimeDelta(column[, maximum_unit])
+formatReadableTimeDelta(column[, maximum_unit, minimum_unit])
 ```
 
 **Arguments**
 
 - `column` — A column with a numeric time delta.
-- `maximum_unit` — Optional. Maximum unit to show. Acceptable values `seconds`, `minutes`, `hours`, `days`, `months`, `years`.
+- `maximum_unit` — Optional. Maximum unit to show.
+  * Acceptable values: `nanoseconds`, `microseconds`, `milliseconds`, `seconds`, `minutes`, `hours`, `days`, `months`, `years`.
+  * Default value: `years`.
+- `minimum_unit` — Optional. Minimum unit to show. All smaller units are truncated.
+  * Acceptable values: `nanoseconds`, `microseconds`, `milliseconds`, `seconds`, `minutes`, `hours`, `days`, `months`, `years`. 
+  * If explicitly specified value is bigger than `maximum_unit`, an exception will be thrown.
+  * Default value: `seconds` if `maximum_unit` is `seconds` or bigger, `nanoseconds` otherwise.  
+
+:::warning
+Sub-second precision may be bad due to the nature of floating-point numbers internal representation. See example below:
+:::
 
-Example:
+``` sql
+SELECT formatReadableTimeDelta(123.3, 'seconds', 'nanoseconds') AS bad_representation;
+SELECT formatReadableTimeDelta(123.4, 'seconds', 'nanoseconds') AS good_representation;
+```
+
+``` text
+┌─bad_representation──────────────────────────────────────────────────┐
+│ 123 seconds, 299 milliseconds, 999 microseconds and 999 nanoseconds │
```

Comment:
This is wrong.
We should use the same logic as `to shortest` in `double-conversion`.

Example:
```
milovidov@milovidov-desktop:~$ clickhouse-local 
ClickHouse local version 23.9.1.1.

milovidov-desktop :) SELECT 123.3

SELECT 123.3

Query id: 832735e7-d22a-4669-8276-16a91ce93aa8

┌─123.3─┐
│ 123.3 │
└───────┘

1 row in set. Elapsed: 0.001 sec. 

milovidov-desktop :) SELECT 123.3::Decimal(10, 10)

SELECT CAST('123.3', 'Decimal(10, 10)')

Query id: 2ec5e8a8-d276-455e-8e45-f1020716fe09

┌─CAST('123.3', 'Decimal(10, 10)')─┐
│                            123.3 │
└──────────────────────────────────┘

1 row in set. Elapsed: 0.001 sec. 

milovidov-desktop :) SELECT CAST(123.3 AS Decimal(10, 10))

SELECT CAST(123.3, 'Decimal(10, 10)')

Query id: 135f276b-a094-4a81-9a85-f1eda5f4139d

┌─CAST(123.3, 'Decimal(10, 10)')─┐
│                          123.3 │
└────────────────────────────────┘

1 row in set. Elapsed: 0.001 sec.
```